esphome/components/ltr_als_ps/ltr_als_ps.cpp

```diff
@@ -0,0 +1,520 @@
+
+    default:
+      break;
+  }
+}
+
+void LTRAlsPsComponent::check_and_trigger_ps_() {
+  static uint32_t last_high_trigger_time{0};
+  static uint32_t last_low_trigger_time{0};
+  uint16_t ps_data = this->read_ps_data_();
+  uint32_t now = millis();
+
+  if (ps_data != this->ps_readings_) {
+    this->ps_readings_ = ps_data;
+    // Higher values - object is closer to sensor
+    if (ps_data > this->ps_threshold_high_ && now - last_high_trigger_time >= this->ps_cooldown_time_s_ * 1000) {
+      last_high_trigger_time = now;
+      ESP_LOGV(TAG, "Proximity high threshold triggered. Value = %d, Trigger level = %d", ps_data,
+               this->ps_threshold_high_);
+      this->on_ps_high_trigger_callback_.call();
+    } else if (ps_data < this->ps_threshold_low_ && now - last_low_trigger_time >= this->ps_cooldown_time_s_ * 1000) {
+      last_low_trigger_time = now;
+      ESP_LOGV(TAG, "Proximity low threshold triggered. Value = %d, Trigger level = %d", ps_data,
+               this->ps_threshold_low_);
+      this->on_ps_low_trigger_callback_.call();
+    }
+  }
+}
+
+bool LTRAlsPsComponent::check_part_number_() {
+  uint8_t manuf_id = this->reg((uint8_t) CommandRegisters::MANUFAC_ID).get();
+  if (manuf_id != 0x05) {  // 0x05 is Lite-On Semiconductor Corp. ID
+    ESP_LOGW(TAG, "Unknown manufacturer ID: 0x%02X", manuf_id);
+    this->mark_failed();
+    return false;
+  }
+
+  // Things getting not really funny here, we can't identify device type by part number ID
+  // ======================== ========= ===== =================
+  // Device                    Part ID   Rev   Capabilities
+  // ======================== ========= ===== =================
+  // Ltr-329/ltr-303            0x0a    0x00  Als 16b
+  // Ltr-553/ltr-556/ltr-556    0x09    0x02  Als 16b + Ps 11b  diff nm sens
+  // Ltr-659                    0x09    0x02  Ps 11b and ps gain
+  //
+  // There are other devices which might potentially work with default settings,
+  // but registers layout is different and we can't use them properly. For ex. ltr-558
+
+  PartIdRegister part_id{0};
+  part_id.raw = this->reg((uint8_t) CommandRegisters::PART_ID).get();
+  if (part_id.part_number_id != 0x0a && part_id.part_number_id != 0x09) {
+    ESP_LOGW(TAG, "Unknown part number ID: 0x%02X. It might not work properly.", part_id.part_number_id);
+    this->status_set_warning();
+    return true;
+  }
+  return true;
+}
+
+void LTRAlsPsComponent::configure_reset_() {
+  ESP_LOGV(TAG, "Resetting");
+
+  AlsControlRegister als_ctrl{0};
+  als_ctrl.sw_reset = true;
+  this->reg((uint8_t) CommandRegisters::ALS_CONTR) = als_ctrl.raw;
+  delay(2);
+
+  uint8_t tries = MAX_TRIES;
+  do {
+    ESP_LOGV(TAG, "Waiting chip to reset");
+    delay(2);
+    als_ctrl.raw = this->reg((uint8_t) CommandRegisters::ALS_CONTR).get();
+  } while (als_ctrl.sw_reset && tries--);  // while sw reset bit is on - keep waiting
+
+  if (als_ctrl.sw_reset) {
+    ESP_LOGW(TAG, "Failed to finalize reset procedure");
```

Comment:
I think I would just say:
```suggestion
    ESP_LOGW(TAG, "Reset timed out");
```